tripal/tripal
# Cannot Sync cvterm-based entities

# Bug Fix

Issue n/a

## Description
Currently, if you create a content type and then immediately try to publish records, it will fail. This is due to the change to `chado_query()` which requires the drupal table to be retrievable via `drupal_get_schema()`. Since the table was just created as part of the create tripal bundle job, it is not in the list returned by `drupal_get_schema()`. To fix this, I cleared the drupal schema cache after creating a new bundle.

## Testing?
1. Create a new Tripal Content Type with base chado table `cvterm`.
2. Run the Tripal job to create it.
3. Publish records.

Not on this branch it will fail with a SQL syntax error due to un-replaced `[chado_bio_data_##]` in the query. On this branch it will work perfectly :-)

NOTE: This PR also fixes a few other minor bugs related to publishing cvterm-based content types. Specifically, adds defaults to validating widget bundle/delta and ensure the bundle can be retrieved for cvterm-based content types in SIO:vocabulary field.
